hashmap/hashmap.mbt

```diff
@@ -101,15 +101,23 @@ pub fn op_set[K : Hash + Eq, V](self : T[K, V], key : K, value : V) -> Unit {
 ///|
 /// Get the value associated with a key.
 pub fn get[K : Hash + Eq, V](self : T[K, V], key : K) -> V? {
-  let hash = key.hash()
+  match self.get_entry_with_hash(key, key.hash()) {
+    Some(entry) => Some(entry.value)
+    None => None
+  }
+}
+
+///|
+fn get_entry_with_hash[K : Hash + Eq, V](
```

Comment:
This is for later optimization on grow. To avoid grow to reuse old entries instead of recreating entries.